src/lazy/text/buffer.rs

```diff
@@ -372,6 +378,111 @@ impl<'data> TextBufferView<'data> {
     fn take_base_16_digits1(self) -> IonMatchResult<'data> {
         take_while1(|b: u8| b.is_ascii_hexdigit())(self)
     }
+
+    /// Matches an Ion float of any syntax
+    fn match_float(self) -> IonParseResult<'data, MatchedFloat> {
+        alt((
+            Self::match_float_special_value,
+            Self::match_float_numeric_value,
+        ))(self)
+    }
```

Comment:
🗺️ This is the entry point for parsing`float`s, which have two kinds of syntax: keyword special values (`+inf`, `-inf`, `nan`), and numeric values. Numeric values come in a big variety of shapes.